lucet-runtime/lucet-runtime-tests/src/guest_fault.rs

```diff
@@ -286,46 +318,50 @@ macro_rules! guest_fault_tests {
-                lucet_runtime::install_lucet_signal_handler();
+                    lucet_runtime::install_lucet_signal_handler();
+                    // call it a few times; it shouldn't matter!
+                    lucet_runtime::install_lucet_signal_handler();
+                    lucet_runtime::install_lucet_signal_handler();
 
-                match inst.run("illegal_instr", &[]) {
-                    Err(Error::RuntimeFault(details)) => {
-                        assert_eq!(details.trapcode, Some(TrapCode::BadSignature));
+                    match inst.run("illegal_instr", &[]) {
+                        Err(Error::RuntimeFault(details)) => {
+                            assert_eq!(details.trapcode, Some(TrapCode::BadSignature));
+                        }
+                        res => panic!("unexpected result: {:?}", res),
                     }
-                    res => panic!("unexpected result: {:?}", res),
-                }
 
-                // after a fault, can reset and run a normal function
-                inst.reset().expect("instance resets");
+                    // after a fault, can reset and run a normal function
+                    inst.reset().expect("instance resets");
 
-                run_onetwothree(&mut inst);
+                    run_onetwothree(&mut inst);
 
-                lucet_runtime::remove_lucet_signal_handler();
-                // call it a few times; it shouldn't matter!
-                lucet_runtime::remove_lucet_signal_handler();
-                lucet_runtime::remove_lucet_signal_handler();
-                lucet_runtime::remove_lucet_signal_handler();
+                    lucet_runtime::remove_lucet_signal_handler();
+                    // call it a few times; it shouldn't matter!
+                    lucet_runtime::remove_lucet_signal_handler();
+                    lucet_runtime::remove_lucet_signal_handler();
+                    lucet_runtime::remove_lucet_signal_handler();
 
-                // just reinstall once and make sure we catch the trap
-                lucet_runtime::install_lucet_signal_handler();
+                    // just reinstall once and make sure we catch the trap
+                    lucet_runtime::install_lucet_signal_handler();
 
-                match inst.run("illegal_instr", &[]) {
-                    Err(Error::RuntimeFault(details)) => {
-                        assert_eq!(details.trapcode, Some(TrapCode::BadSignature));
+                    match inst.run("illegal_instr", &[]) {
+                        Err(Error::RuntimeFault(details)) => {
+                            assert_eq!(details.trapcode, Some(TrapCode::BadSignature));
+                        }
+                        res => panic!("unexpected result: {:?}", res),
                     }
-                    res => panic!("unexpected result: {:?}", res),
-                }
+
+                    lucet_runtime::remove_lucet_signal_handler();
```

Comment:
:bug:

Reply:
this should be a no-op, since we've opted to manually control signal handling, right?

Reply:
It is for this test... but not for ones that might run after this one :imp: